crypto/src/main/java/net/consensys/cava/crypto/sodium/SecretBox.java

```diff
@@ -193,7 +193,7 @@ public Bytes bytes() {
   /**
    * A SecretBox nonce.
    */
-  public static final class Nonce {
+  public static final class Nonce implements AutoCloseable {
```

Comment:
I'm not sure I agree with making a Nonce "auto closeable". It's not a resource, so it feels odd to treat it like one. We could certainly make it implement `Destroyable`, just like the secret keys.

However, whatever we do, it must be applied consistently across all cava-sodium types.

Reply:
@cleishm - I see where you're coming from. I've pushed a new commit to this PR to implement Destroyable instead.

There's currently a couple of other classes (below) that implement `AutoCloseable` that aren't 'resources' (like sockets) but on searching, it looks like the majority implement `Destroyable` like you suggested so I'll follow that convention.

```
public final class AES256GCM implements AutoCloseable {
```
```
public final class Box implements AutoCloseable {
```
Perhaps these can be made to implement 'destroyable' also in a separate PR instead.

Reply:
It's a different flow. You open a secret box and close it - you don't destroy it. It's more the pointers of the keys and nonces generated along the way we need to clean up.